src/RA_Dlg_MemBookmark.cpp

```diff
@@ -650,19 +647,22 @@ void Dlg_MemBookmark::WriteFrozenValue(const MemBookmark & Bookmark)
         default:
             break;
     }
-
-    char buffer[32];
-    sprintf_s(buffer, sizeof(buffer), "%0*x", width, Bookmark.Value());
-
-    for (unsigned int i = 0; i < strlen(buffer); i++)
+    
+    if (auto buffer{ std::make_unique<char[]>(32) }; sprintf_s(buffer.get(), 32, "%0*x", width, Bookmark.Value()) != -1)
     {
-        c = buffer[i];
-        n = (c >= 'a') ? (c - 'a' + 10) : (c - '0');
-        MemoryViewerControl::editData(addr, (i % 2 != 0), n);
+        std::string str{ buffer.release() };
```

Comment:
From what I've read, std;:string uses RAII and owns a pointer to string. I've also read std::string_view (immutable std::string) does not express ownership. I'll just change it to get and it should release it by itself.

Reply:
Internally, it may own a pointer to a string, but this is just calling the "std::string(const char*)" constructor. That copies from the pointer without taking ownership.